### DISCUSSION TITLE
Add IsAnimated property (alternatively: FrameCount) to ImageInfo

### DISCUSSION THREAD

**Discussion by zotanmew** [SixLabors/ImageSharp#2777] [Ideas] [answered]
This would make special handling of animated images (e.g., to skip processing them) trivial.

**Comment by zotanmew** [comment]
I wrote some code for this: https://iceshrimp.dev/iceshrimp/ImageSharp/commit/7ab7ff4b6a74acd4f32381d93d1e8e5159a3f6dc

Happy to submit a PR if wanted.

**Comment by JimBobSquarePants** [comment]
Hi @zotanmew Thanks for the suggestion but unfortunately the presence of multiple frames does not automatically mean that the image is animated. Container formats like Tiff for example can have multiple frames.

**Comment by zotanmew** [comment]
@JimBobSquarePants How about `FrameCount`? Would be similarly useful without that issue. I fully understand if you aren't interested in upstreaming this, however.

**Reply by JimBobSquarePants** [reply, marked as answer]
V4 will contain an explicit `FrameCount` property on `ImageInfo ` as part of #2751
